### PR TITLE
Setup base GraphQL

### LINKED ISSUE #26: Add GraphQL
## Why
- Try to implement GraphQL in Rails project
- It can open many new opportunities to practice for the mobile/client side

### PR DESCRIPTION
**Pull request by hoangmirs**
## What happened
- Add GraphQL support to project
- Setup the base for GraphQL
- Add example
- Close #26 
 
## Insight
- Use [graphql](https://graphql-ruby.org/) gem, ignore installing graphiql(mount the GraphQL IDE in the app) as it's often used in the development, and we have some endpoints that require authentication, and it doesn't have the place to put headers, we can add it but will take more time
- We can use [GraphiQL.app](https://github.com/skevy/graphiql-app) instead
- Add the `BaseResolver`, `resolvers` folder to reduce complexity in `query_type`
- Add the `BaseMutation`, `mutations` folder to reduce complexity in `mutation_type`
- Move all base types into `base` folder
- Put `graphql_controller` in `api` folder as it's still an API
- Add examples for query and mutation
- I planned to move the graphql part to an engine, but this is a small app, it's still fine to keep it in main app
- [GraphQL-rules](https://graphql-rules.com/rules/naming) recommend using `cameCase` for fields and arguments. Github and Spotify also use `camelCase` for their GraphQL endpoints, so I keep using `cameCase` for GraphQL as well. I still keep the OAuth2 API using `snake_case`

## Proof Of Work
- Tests should be passed
- Can access `POST /api/v1/graphql` to test

### CODE REVIEW

File: app/controllers/api/graphql_controller.rb [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+module API
+  class GraphqlController < ActionController::API
+    # If accessing from outside this domain, nullify the session
+    # This allows for outside API access while preventing CSRF attacks,
+    # but you'll have to authenticate your user separately
+    # protect_from_forgery with: :null_session
+
+    def execute
+      render json: result
+    rescue StandardError => e
```
Comment (team-nimblehq) [inline app/controllers/api/graphql_controller.rb:12]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/46106ccd13ad8d3333b75e32a92764a807848798/app/controllers/api/graphql_controller.rb#L12"></span>Has the variable name ‘e’</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/graphql_controller.rb [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+    end
+
+    # Handle form data, JSON body, or a blank value
+    def ensure_hash(ambiguous_param)
+      case ambiguous_param
+      when String
+        ambiguous_param.present? ? ensure_hash(JSON.parse(ambiguous_param)) : {}
+      when Hash, ActionController::Parameters
+        ambiguous_param
+      when nil
```
Comment (team-nimblehq) [inline app/controllers/api/graphql_controller.rb:38]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/46106ccd13ad8d3333b75e32a92764a807848798/app/controllers/api/graphql_controller.rb#L38"></span>Performs a nil-check</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/graphql_controller.rb [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        {}
+      else
+        raise ArgumentError, "Unexpected parameter: #{ambiguous_param}"
+      end
+    end
+
+    def handle_error_in_development(exception)
+      logger.error exception.message
```
Comment (team-nimblehq) [inline app/controllers/api/graphql_controller.rb:46]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/46106ccd13ad8d3333b75e32a92764a807848798/app/controllers/api/graphql_controller.rb#L46"></span>Refers to ‘exception’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/graphql_controller.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+      logger.error exception.backtrace.join("\n")
```
Comment (team-nimblehq) [inline app/controllers/api/graphql_controller.rb:47]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/46106ccd13ad8d3333b75e32a92764a807848798/app/controllers/api/graphql_controller.rb#L47"></span>Refers to ‘exception’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/graphql_controller.rb [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+
+      render json: {
+        errors: [{ message: exception.message, backtrace: exception.backtrace }],
```
Comment (team-nimblehq) [inline app/controllers/api/graphql_controller.rb:50]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/46106ccd13ad8d3333b75e32a92764a807848798/app/controllers/api/graphql_controller.rb#L50"></span>Refers to ‘exception’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: app/graphql/queries/survey_query.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+# frozen_string_literal: true
+
+module Queries
+  class SurveyQuery < BaseQuery
+    type [Types::SurveyType], null: false
+    description 'Survey list'
+
+    def resolve
```
Comment (team-nimblehq) [inline app/graphql/queries/survey_query.rb:8]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/46106ccd13ad8d3333b75e32a92764a807848798/app/graphql/queries/survey_query.rb#L8"></span>Doesn’t depend on instance state (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: app/graphql/types/base/field.rb [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+
+module Types
+  module Base
+    class Field < GraphQL::Schema::Field
+      argument_class Types::Base::Argument
+
+      # Add `null: false` and `camelize: false` which provide default values
+      # in case the caller doesn't pass anything for those arguments.
+      # **kwargs is a catch-all that will get everything else
+      def initialize(*args, null: false, camelize: false, **kwargs, &block)
```
Comment (team-nimblehq) [inline app/graphql/types/base/field.rb:11]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/46106ccd13ad8d3333b75e32a92764a807848798/app/graphql/types/base/field.rb#L11"></span>Has boolean parameter ‘camelize’</td>
    </tr>
  </tbody>
</table>

Comment (team-nimblehq) [inline app/graphql/types/base/field.rb:11]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/46106ccd13ad8d3333b75e32a92764a807848798/app/graphql/types/base/field.rb#L11"></span>Has boolean parameter ‘null’</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/v1/graphql_controller.rb [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+  module V1
+    class GraphqlController < ActionController::API
+      # If accessing from outside this domain, nullify the session
+      # This allows for outside API access while preventing CSRF attacks,
+      # but you'll have to authenticate your user separately
+      # protect_from_forgery with: :null_session
+
+      def execute
+        render json: result
+      rescue StandardError => e
```
Comment (team-nimblehq) [inline app/controllers/api/v1/graphql_controller.rb:13]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/bde03b9355743c3ad209800a745b8cc6547ad670/app/controllers/api/v1/graphql_controller.rb#L13"></span>Has the variable name ‘e’</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/v1/graphql_controller.rb [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+      end
+
+      # Handle form data, JSON body, or a blank value
+      def ensure_hash(ambiguous_param)
+        case ambiguous_param
+        when String
+          ambiguous_param.present? ? ensure_hash(JSON.parse(ambiguous_param)) : {}
+        when Hash, ActionController::Parameters
+          ambiguous_param
+        when nil
```
Comment (team-nimblehq) [inline app/controllers/api/v1/graphql_controller.rb:39]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/bde03b9355743c3ad209800a745b8cc6547ad670/app/controllers/api/v1/graphql_controller.rb#L39"></span>Performs a nil-check</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/v1/graphql_controller.rb [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+          {}
+        else
+          raise ArgumentError, "Unexpected parameter: #{ambiguous_param}"
+        end
+      end
+
+      def handle_error_in_development(exception)
+        logger.error exception.message
```
Comment (team-nimblehq) [inline app/controllers/api/v1/graphql_controller.rb:47]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/bde03b9355743c3ad209800a745b8cc6547ad670/app/controllers/api/v1/graphql_controller.rb#L47"></span>Refers to ‘exception’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/v1/graphql_controller.rb [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+        logger.error exception.backtrace.join("\n")
```
Comment (team-nimblehq) [inline app/controllers/api/v1/graphql_controller.rb:48]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/bde03b9355743c3ad209800a745b8cc6547ad670/app/controllers/api/v1/graphql_controller.rb#L48"></span>Refers to ‘exception’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/v1/graphql_controller.rb [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+
+        render json: {
+          errors: [{ message: exception.message, backtrace: exception.backtrace }],
```
Comment (team-nimblehq) [inline app/controllers/api/v1/graphql_controller.rb:51]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/bde03b9355743c3ad209800a745b8cc6547ad670/app/controllers/api/v1/graphql_controller.rb#L51"></span>Refers to ‘exception’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: app/graphql/queries/survey_query.rb [hunk 13]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Queries
+  class SurveyQuery < BaseQuery
+    description 'Survey list'
+
+    type [Types::SurveyType], null: false
+
+    def resolve
```
Comment (team-nimblehq) [inline app/graphql/queries/survey_query.rb:9]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/080830d571ead7083d5b15ab6138b613fcba3462/app/graphql/queries/survey_query.rb#L9"></span>Doesn’t depend on instance state (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/v1/application_controller.rb [hunk 14]
```diff
@@ -25,6 +25,22 @@ def render_error(detail:, source: nil, meta: nil, status: :unprocessable_entity,
 
         render json: { errors: errors }, status: status
       end
+
+      def doorkeeper_unauthorized_render_options(error: nil)
```
Comment (team-nimblehq) [inline app/controllers/api/v1/application_controller.rb:29]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/59338f029e412186a0a6dc405561dc7f2a762a99/app/controllers/api/v1/application_controller.rb#L29"></span>Remove unused methods (api::v1::applicationcontroller#doorkeeper_unauthorized_render_options)</td>
    </tr>
  </tbody>
</table>

Comment (team-nimblehq) [inline app/controllers/api/v1/application_controller.rb:29]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/59338f029e412186a0a6dc405561dc7f2a762a99/app/controllers/api/v1/application_controller.rb#L29"></span>Method has too many lines. [12/10]</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/v1/graphql_controller.rb [hunk 15]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    class GraphqlController < ApplicationController
+      def create
+        render json: result
+      rescue StandardError => e
```
Comment (team-nimblehq) [inline app/controllers/api/v1/graphql_controller.rb:8]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/59338f029e412186a0a6dc405561dc7f2a762a99/app/controllers/api/v1/graphql_controller.rb#L8"></span>Has the variable name ‘e’</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/v1/graphql_controller.rb [hunk 16]
```diff
@@ -0,0 +1,51 @@
...
+      end
+
+      # Handle form data, JSON body, or a blank value
+      def ensure_hash(ambiguous_param)
+        case ambiguous_param
+        when String
+          ambiguous_param.present? ? ensure_hash(JSON.parse(ambiguous_param)) : {}
+        when Hash, ActionController::Parameters
+          ambiguous_param
+        when nil
```
Comment (team-nimblehq) [inline app/controllers/api/v1/graphql_controller.rb:33]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/59338f029e412186a0a6dc405561dc7f2a762a99/app/controllers/api/v1/graphql_controller.rb#L33"></span>Performs a nil-check</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/v1/graphql_controller.rb [hunk 17]
```diff
@@ -0,0 +1,51 @@
...
+          {}
+        else
+          raise ArgumentError, "Unexpected parameter: #{ambiguous_param}"
+        end
+      end
+
+      def handle_error_in_development(exception)
+        logger.error exception.message
```
Comment (team-nimblehq) [inline app/controllers/api/v1/graphql_controller.rb:41]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/59338f029e412186a0a6dc405561dc7f2a762a99/app/controllers/api/v1/graphql_controller.rb#L41"></span>Refers to ‘exception’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/v1/graphql_controller.rb [hunk 18]
```diff
@@ -0,0 +1,51 @@
...
+        logger.error exception.backtrace.join("\n")
```
Comment (team-nimblehq) [inline app/controllers/api/v1/graphql_controller.rb:42]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/59338f029e412186a0a6dc405561dc7f2a762a99/app/controllers/api/v1/graphql_controller.rb#L42"></span>Refers to ‘exception’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: app/controllers/api/v1/graphql_controller.rb [hunk 19]
```diff
@@ -0,0 +1,51 @@
...
+
+        render json: {
+          errors: [{ message: exception.message, backtrace: exception.backtrace }],
```
Comment (team-nimblehq) [inline app/controllers/api/v1/graphql_controller.rb:45]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/59338f029e412186a0a6dc405561dc7f2a762a99/app/controllers/api/v1/graphql_controller.rb#L45"></span>Refers to ‘exception’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: app/graphql/resolvers/survey_resolver.rb [hunk 20]
```diff
@@ -0,0 +1,13 @@
+# frozen_string_literal: true
+
+module Resolvers
+  class SurveyResolver < BaseResolver
+    description 'Survey list'
+
+    type [Types::SurveyType], null: false
+
+    def resolve
```
Comment (team-nimblehq) [inline app/graphql/resolvers/survey_resolver.rb:9]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/59338f029e412186a0a6dc405561dc7f2a762a99/app/graphql/resolvers/survey_resolver.rb#L9"></span>Doesn’t depend on instance state (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

**Comment by team-nimblehq** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><a href="https://github.com/nimblehq/nimble-survey-web/blob/59338f029e412186a0a6dc405561dc7f2a762a99/app/controllers/api/v1/application_controller.rb#L16">app/controllers/api/v1/application_controller.rb#L16</a> - Remove unused methods (api::v1::applicationcontroller#render_error)</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: app/controllers/api/v1/graphql_controller.rb [hunk 21]
```diff
@@ -0,0 +1,51 @@
+# frozen_string_literal: true
+
+module API
+  module V1
+    class GraphqlController < ApplicationController
+      def create
+        render json: result
+      rescue StandardError => e
+        raise e unless Rails.env.development?
```
Comment (ankitkalia1195) [inline app/controllers/api/v1/graphql_controller.rb:9]: @hoangmirs Why do we need to render json error in development, can we not just see the logs and tell the error?

Comment (hoangmirs) [reply app/controllers/api/v1/graphql_controller.rb:9]: This comes from the graphql gem, we can remove it

File: app/controllers/api/v1/graphql_controller.rb [hunk 22]
```diff
@@ -0,0 +1,37 @@
...
+      end
+
+      # Handle form data, JSON body, or a blank value
+      def ensure_hash(ambiguous_param)
+        case ambiguous_param
+        when String
+          ambiguous_param.present? ? ensure_hash(JSON.parse(ambiguous_param)) : {}
+        when Hash, ActionController::Parameters
+          ambiguous_param
+        when nil
```
Comment (team-nimblehq) [inline app/controllers/api/v1/graphql_controller.rb:29]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/e9f54d3527584ffca870370963091e7829cac8bd/app/controllers/api/v1/graphql_controller.rb#L29"></span>Performs a nil-check</td>
    </tr>
  </tbody>
</table>

File: Gemfile [hunk 23]
```diff
@@ -14,6 +14,7 @@ gem 'sidekiq' # background processing for Ruby
 gem 'bootsnap', require: false # Reduces boot times through caching; required in config/boot.rb
 gem 'i18n-js', '3.5.1' # A library to provide the I18n translations on the Javascript
 gem 'fast_jsonapi' # A lightning fast JSON:API serializer for Ruby Objects
+gem 'graphql' # Support GraphQL
```
Comment (olivierobert) [inline Gemfile:17]: How about changing the description to "Ruby implementation of GraphQL" as usually, we rely on the project description. Minot thing though.

File: config/routes.rb [hunk 24]
```diff
@@ -19,6 +19,7 @@
         resources :passwords, only: :create
       end
 
+      resources :graphql, only: :create
```
Comment (olivierobert) [inline config/routes.rb:22]: Not sure about the route location as 1) `api/v1` is meant only for the REST API and 2) GraphQL does not handle versioning with paths but with deprecated fields. Having simply the endpoint `/graphql` would do as done similarly by [Github](https://developer.github.com/v4/guides/forming-calls/#the-graphql-endpoint).

**Review by olivierobert — COMMENTED** [review]
Looks good to me with some minor comments. Make sure to clean up all test controller and data afterwards.

File: app/controllers/graphql_controller.rb [hunk 25]
```diff
@@ -0,0 +1,33 @@
...
+  end
+
+  # Handle form data, JSON body, or a blank value
+  def ensure_hash(ambiguous_param)
+    case ambiguous_param
+    when String
+      ambiguous_param.present? ? ensure_hash(JSON.parse(ambiguous_param)) : {}
+    when Hash, ActionController::Parameters
+      ambiguous_param
+    when nil
```
Comment (team-nimblehq) [inline app/controllers/graphql_controller.rb:27]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/f8993aea2ddd1837b0cc58c7f0ee11168f125be9/app/controllers/graphql_controller.rb#L27"></span>Performs a nil-check</td>
    </tr>
  </tbody>
</table>

File: app/controllers/concerns/api/v1/doorkeeper_authentication.rb [hunk 26]
```diff
@@ -0,0 +1,30 @@
...
+      def current_user
+        User.find(doorkeeper_token.resource_owner_id) if doorkeeper_token
+      end
+
+      def doorkeeper_unauthorized_render_options(error: nil)
+        return unless error
+
+        {
+          json: {
+            errors: build_errors(detail: error.description, source: error.state, code: error.name)
```
Comment (team-nimblehq) [inline app/controllers/concerns/api/v1/doorkeeper_authentication.rb:24]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/ddfec4932dbb11ae78f67f52b257b3efda688cc1/app/controllers/concerns/api/v1/doorkeeper_authentication.rb#L24"></span>Refers to ‘error’ more than self (maybe move it to another class?)</td>
    </tr>
  </tbody>
</table>

File: app/controllers/graphql_controller.rb [hunk 27]
```diff
@@ -0,0 +1,35 @@
...
```
Comment (team-nimblehq) [inline app/controllers/graphql_controller.rb:29]: <table data-meta="generated_by_danger">
  <tbody>
    <tr>
      <td>:warning:</td>
      <td width="100%" data-sticky="false"><span data-href="https://github.com/nimblehq/nimble-survey-web/blob/ddfec4932dbb11ae78f67f52b257b3efda688cc1/app/controllers/graphql_controller.rb#L29"></span>Performs a nil-check</td>
    </tr>
  </tbody>
</table>

File: config/routes.rb (hunk 24 above)
Comment (hoangmirs) [reply config/routes.rb]: Fixed in https://github.com/nimblehq/nimble-survey-web/pull/27/commits/1f8c00fe60f1730685034b078441870311349a29

**Review by olivierobert — APPROVED** [review]

**Pull request merged**